IPIP/0000-gateway-json-cbor-response-format.md

```diff
@@ -0,0 +1,79 @@
+# IPIP 0000: Gateway JSON and TAR Response Formats
+
+- Start Date: 2022-10-07
+- Related Issues:
+  - [ipfs/in-web-browsers/issues/182]
+  - [ipfs/specs/pull/328]
+  - [ipfs/kubo/issues/8823]
+  - [ipfs/kubo/pull/9335]
+  - [ipfs/go-ipfs/issues/7552]
+
+## Summary
+
+Add [DAG-JSON], [DAG-CBOR], JSON and CBOR response formats to the [HTTP Gateway](../http-gateways/).
+
+## Motivation
+
+Currently, the gateway supports requesting data in the [DAG-PB], RAW, and [CAR]
+formats. However, it should be possible to download everything over the gateway
+in order to unlock the potentials of the [IPLD Data Model][ipld-data-model].
+
+For example, DAG-PB has a [logical format][dag-pb-format] which makes it possible
+to represent a DAG-PB directory as a [DAG-JSON] document. This means that, if we
+support DAG-JSON in the gateway, then we would support
+[JSON responses for directory listings][ipfs/go-ipfs/issues/7552], which has been
+requested by our users in the past.
+
+In addition, this functionality is already present on the current Kubo CLI. By
+bringing it to the gateways, we provide users with more power when it comes
+to traversing IPLD data.
+
+## Detailed design
+
+The solution is to allow the Gateway to support serializing an IPLD model
+representation as [DAG-JSON], [DAG-CBOR], JSON and CBOR by requesting them
+using either the `Accept` HTTP header or the `format` URL query.
+
+## Test fixtures
+
+TODO
+
+## Design rationale
+
+The current gateway already supports different response formats via the
+`Accept` HTTP header and the `format` URL query. This IPIP proposes adding
+one more supported format to that list.
+
+### User benefit
+
+The user benefits from this change as they will now be able to retrieve
+content encoded in the traversable DAG-JSON and DAG-CBOR formats. This is
+something that has been [requested before][ipfs/go-ipfs/issues/7552].
+
+### Compatibility
+
+This IPIP is backwards compatible.
+
+### Security
+
+TODO
+
+### Alternatives
+
+TODO
```

Comment:
1. Describe the result of doing nothing – IPFS use is stuck around files and directories represented with UnixFS, and IPLD potential remains locked due to artificial barrier created by gateways
2. Note that introducing  DAG  variants without  having generic JSON and CBOR would produce bad UX – ability to retrieve DAG-JSON as `application/json` is  important for interop with web browsers and other tools which expect specific content type.